src/primitives.js

```diff
@@ -1217,17 +1225,17 @@ export default {
 					color: { value: edgeForm.color ?? [0, 0, 0] }
 				},
 				vertexShader: `
-					attribute vec3 offset;
+					attribute vec3 polyhedronCenter;
 
 					void main() {
-						gl_Position = projectionMatrix * modelViewMatrix * vec4(vec3(position) + offset, 1.0);
+						gl_Position = projectionMatrix * modelViewMatrix * vec4(position + polyhedronCenter, 1);
 					}
 				`,
 				fragmentShader: `
 					uniform vec3 color;
 
 					void main() {
-						gl_FragColor = vec4(color, 1.0);
+						gl_FragColor = vec4(color, 1);
 					}
 				`
 			})
```

Comment:
At 1.2K lines it might be time to start considering breaking this up onto smaller files. One kind of breakout might be inherently 1D and 2D objects vs 3D objects. Or things that use a similar kind of shader.